README.md

```diff
@@ -1,22 +1,27 @@
 
  There are two primary approaches used while testing software: Functional Testing and Non-functional Testing. Functional testing focuses on the functionality of a software product. It helps to understand whether the software is performing as needed. Each functionality is tested by providing the input, calculating the output, and comparing the actual result to the anticipated value. While non-functional testing is a sort of software testing used to evaluate the software's non-functional characteristics such as reliability, performance, and accountability. Testing, both functional and non-functional, is necessary for newly produced software. While non-functional testing examines an application's capacity to operate in an external environment, functional testing examines the accuracy of internal functions. ( [Unit testing- Javatpoint](https://www.javatpoint.com/types-of-software-testing) )
 
  In this article, we are concentrating on unit testing, a type of functional testing.
  
  ## What is Unit testing? 
-Unit testing is a form of software testing that focuses on validating individual units or components of a software application and confirming that each unit functions as intended. It is considered the first phase of functional testing in the software development life cycle. It ensures that it has no impact on any other parts of your code than the component or unit under test by isolating them from the rest of your code. As a result, it isolates a single functional component, guarantees correctness, and confirms that there are no errors, enabling speedy bug fixes before merging them into the main code. It is a type of "white box" testing. It is frequently accomplished by writing test cases that test specific functions, methods, and classes. Test cases can be written in plain English, even though they are usually written in pseudocode (implemented in a language like Java or JavaScript).
+Unit testing is a form of software testing that focuses on validating individual units or components of a software application and confirming that each unit functions as intended ([ref?]()). It is considered the first phase of functional testing in the software development life cycle ([ref?]()). It ensures that it has no impact on any other parts of your code than the component or unit under test by isolating them from the rest of your code ([ref?]()). As a result, it isolates a single functional component, guarantees correctness, and confirms that there are no errors _\[from: https://www.lambdatest.com/learning-hub/unit-testing\]_, enabling speedy bug fixes before merging them into the main code ([ref?]()). It is a type of "white box" testing. It is frequently accomplished by writing test cases that test specific functions, methods, and classes ([ref?]()). Test cases can be written in plain English, even though they are usually written in pseudocode (implemented in a language like Java or JavaScript) ([ref?]()).
```

Comment:
You are missing references in this paragraph (and in many others). I've put in a reference request for each sentence; however, you could get away with a single reference only at the end of the paragraph, if _all_ the information in this paragraph came from the same source. Otherwise, you'd need references on each sentence that comes from a different source.

Reply:
Also, there is a section in the middle that is a copy (with minor changes) from the url provided in my changes. You've used that particular url as a reference elsewhere, but not here, and I'm not sure why. Either way, the wording is too close to the original, and without any reference at all, this would appear to be plagiarism.